File: ZubaleService/src/main/java/com/zubale/model/Quote.java
package com.zubale.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Quote {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	Long id;
	String quoteString;
	Long votes = 0l;

	public Quote(Long id,
	String quoteString,
	Long votes) {
		this.id = id;
		this.quoteString = quoteString;
		this.votes = votes;
	}
	
	public Quote() {
		
	}
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Long getVotes() {
		return votes;
	}
	public void setVotes(Long votes) {
		this.votes = votes;
	}

	public String getQuoteString() {
		return quoteString;
	}

	public void setQuoteString(String quoteString) {
		this.quoteString = quoteString;
	}
	
	
}

File: src/main/kotlin/com/pulpos/quotes/model/Quote.kt
package com.pulpos.quotes.model

import com.fasterxml.jackson.annotation.*
import java.time.LocalDate
import javax.persistence.*
import javax.validation.constraints.NotBlank

@Entity
data class Quote (
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Long = 0,

        @get: NotBlank
        var text: String = "",

        val date: LocalDate? = null,

        var votes: Long = 0,

        var author: String = ""

) {

        @ManyToOne
        lateinit var user: User

        constructor(text: String, date: LocalDate, author: String, user : User) : this(text = text, date = date, author = author) {
                this.user = user
        }
}
